File: apps/api/app/schemas/ai_dashboard.py
from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel


class AiDashboardKpi(BaseModel):
    label: str
    value: str
    sub: str
    pct: Optional[int] = None
    alert: Optional[bool] = None
    up: Optional[bool] = None


class AiDashboardTokenDay(BaseModel):
    day: str
    tokens: int


class AiDashboardAgentActivity(BaseModel):
    agent: str
    runs: int


class AiDashboardData(BaseModel):
    kpis: List[AiDashboardKpi]
    token_usage_chart: List[AiDashboardTokenDay]
    agent_activity_chart: List[AiDashboardAgentActivity]


class AiDashboardResponse(BaseModel):
    data: AiDashboardData

File: apps/web/src/lib/api/ai-dashboard.ts
export type AiDashboardKpi = {
  label: string;
  value: string;
  sub: string;
  pct?: number;
  alert?: boolean;
  up?: boolean;
};

export type AiDashboardTokenDay = {
  day: string;
  tokens: number;
};

export type AiDashboardAgentActivity = {
  agent: string;
  runs: number;
};

export type ApiAiDashboardData = {
  kpis: AiDashboardKpi[];
  token_usage_chart: AiDashboardTokenDay[];
  agent_activity_chart: AiDashboardAgentActivity[];
};

export type ApiAiDashboardResponse = {
  data: ApiAiDashboardData;
};

export function apiDashboardToView(data: ApiAiDashboardData) {
  return {
    kpis: data.kpis,
    tokenUsageChart: data.token_usage_chart.map((row) => ({
      day: row.day,
      tokens: row.tokens,
    })),
    agentActivityChart: data.agent_activity_chart.map((row) => ({
      agent: row.agent,
      runs: row.runs,
    })),
  };
}
